File: native/engine/ios/Plugins/CallbackContext.m
#import "CallbackContext.h"
#include "JsbCall.h"
@implementation CallbackContext

- (instancetype)initWithCallbackId:(NSString *)callbackId {
    self = [super init];
    if (self) {
        self._callbackId = callbackId;
    }
    return self;
}

- (void)successWithParams:(NSString *)params {
    [JsbCall successCall:self._callbackId params:params];
}

- (void)failureWithParams:(NSString *)params {
    [JsbCall failureCall:self._callbackId params:params];
}

- (void)successWithJSONObject:(NSDictionary *)obj {
    [JsbCall successCall:self._callbackId params:[self jsonStringFromJSONObject:obj]];
}

- (void)failureWithJSONObject:(NSDictionary *)obj {
    [JsbCall failureCall:self._callbackId params:[self jsonStringFromJSONObject:obj]];
}

- (NSString *)jsonStringFromJSONObject:(NSDictionary *)obj {
    NSError *error;
    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:obj options:0 error:&error];
    if (!jsonData) {
        NSLog(@"Error converting JSONObject to JSON string: %@", error);
        return @"";
    } else {
        return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
    }
}

@end

File: native/engine/android/app/src/com/brix/jsb/CallbackContext.java
package com.brix.jsb;

import org.json.JSONObject;

public class CallbackContext {

    private String mCallbackId = null;

    public CallbackContext(String callbackId) {
        mCallbackId = callbackId;
    }

    public void success(String params) {
        JsbCall.successCall(mCallbackId, params);
    }

    public void failure(String params) {
        JsbCall.failureCall(mCallbackId, params);
    }
    public void success(JSONObject obj){
        JsbCall.successCall(mCallbackId, obj.toString());
    }
    public void failure(JSONObject obj) {
        JsbCall.failureCall(mCallbackId, obj.toString());
    }
}
